### PR TITLE
src: fix qt 5.15 warnings

### PR DESCRIPTION
**Pull request by selsta**

### CODE REVIEW

File: src/qt/utils.cpp [hunk 1]
```diff
@@ -68,7 +69,12 @@ QByteArray fileOpen(QString path) {
 bool fileWrite(QString path, QString data) {
     QFile file(path);
     if(file.open(QIODevice::WriteOnly)){
-        QTextStream out(&file); out << data << endl;
+        QTextStream out(&file);
+#if QT_VERSION >= QT_VERSION_CHECK(5, 14, 0)
+        out << data << Qt::endl;
+#else
+        out << data << endl;
+#endif
```
Comment (xiphon) [inline src/qt/utils.cpp:77]: Use '\n' instead

File: src/libwalletqt/TransactionHistory.cpp [hunk 2]
```diff
@@ -143,7 +148,11 @@ bool TransactionHistory::TransactionHistory::locked() const
 TransactionHistory::TransactionHistory(Monero::TransactionHistory *pimpl, QObject *parent)
     : QObject(parent), m_pimpl(pimpl), m_minutesToUnlock(0), m_locked(false)
 {
+#if QT_VERSION >= QT_VERSION_CHECK(5, 14, 0)
+    m_firstDateTime = QDate(2014, 4, 18).startOfDay();
+#else
     m_firstDateTime  = QDateTime(QDate(2014, 4, 18)); // the genesis block
+#endif
```
Comment (xiphon) [inline src/libwalletqt/TransactionHistory.cpp:155]: Could you post the warning?

File: src/qt/utils.cpp (hunk 1 above)
Comment (selsta) [reply src/qt/utils.cpp:77]: No flush is okay?

File: src/libwalletqt/TransactionHistory.cpp (hunk 2 above)
Comment (selsta) [reply src/libwalletqt/TransactionHistory.cpp:155]:
```
../../Qt/5.15.0/clang_64/lib/QtCore.framework/Headers/qdatetime.h:296:5: note: 'QDateTime' has been explicitly marked deprecated here
    QT_DEPRECATED_X("Use QDate::startOfDay()") explicit QDateTime(const QDate &);
    ^
```

File: src/qt/utils.cpp (hunk 1 above)
Comment (xiphon) [reply src/qt/utils.cpp:77]: Yep

Comment (selsta) [reply src/qt/utils.cpp]: updated

File: src/libwalletqt/TransactionHistory.cpp (hunk 2 above)
Comment (xiphon) [reply src/libwalletqt/TransactionHistory.cpp:155]: I see. Wonder they don't mention this in the docs.

**Pull request merged**